textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/relation/RelationManager.java

```diff
@@ -0,0 +1,363 @@
+            insertTupleToDirectory(schemaCatalogStore, LuceneAnalyzerConstants.getStandardAnalyzer(), tuple);
+        }
+        
+    }
+
+    // drop a table
+    public void deleteTable(String tableName) throws StorageException {
+        Query catalogTableNameQuery = new TermQuery(new Term(CatalogConstants.COLLECTION_NAME, tableName));
+        
+        DataWriter collectionCatalogWriter = new DataWriter(CatalogConstants.COLLECTION_CATALOG_DATASTORE, 
+                LuceneAnalyzerConstants.getStandardAnalyzer());
+        collectionCatalogWriter.deleteTuple(catalogTableNameQuery);
+        
+        DataWriter schemaCatalogWriter = new DataWriter(CatalogConstants.SCHEMA_CATALOG_DATASTORE,
+                LuceneAnalyzerConstants.getStandardAnalyzer());
+        schemaCatalogWriter.deleteTuple(catalogTableNameQuery);
+    }
+
+    // insert a tuple to a table, returns the ID field
+    public IDField insertTuple(String tableName, ITuple tuple) throws StorageException {
+        if (! checkTableExistence(tableName)) {
+            throw new StorageException(String.format("Table %s doesn't exist.", tableName));
+        }
+        try {
+            DataStore tableDataStore = getTableDataStore(tableName);
+            String tableAnalyzerString = getTableAnalyzer(tableName);
+            Analyzer luceneAnalyzer = LuceneAnalyzerConstants.getLuceneAnalyzer(tableAnalyzerString);
+
+            return insertTupleToDirectory(tableDataStore, luceneAnalyzer, tuple);
+        } catch (DataFlowException e) {
+            throw new StorageException(e);
+        }
+    }
+    
+    private IDField insertTupleToDirectory(IDataStore dataStore, Analyzer luceneAnalyzer, ITuple tuple) throws StorageException {
+        String tableDirectory = dataStore.getDataDirectory();
+        Schema tableSchema = dataStore.getSchema();
+        IDField idField;
+                
+        if (! tableSchema.containsField(SchemaConstants._ID)) {
+            tableSchema = Utils.getSchemaWithID(tableSchema);
+        }
+        
+        ITuple insertionTuple = tuple;
+        // add "_id" to schema, and add ID field to tuple
+        if (! insertionTuple.getSchema().containsField(SchemaConstants._ID)) {
+            idField = new IDField(UUID.randomUUID().toString());
+            insertionTuple = getTupleWithID(tuple, idField);
+        } else {
+            idField = (IDField) tuple.getField(SchemaConstants._ID);
+        }
+
+        if (! tableSchema.equals(insertionTuple.getSchema())) {
+            throw new StorageException("Tuple's schema is inconsistent with table schema.");
+        }
+        
+        DataWriter dataWriter = new DataWriter(new DataStore(tableDirectory, tableSchema), luceneAnalyzer);
+        dataWriter.insertTuple(insertionTuple);
+
+        return idField;    
+    }
+
+    // delete a tuple by its id
+    public void deleteTuple(String tableName, IField idValue) throws StorageException {
+        try {
```

Comment:
Add a comment to the codebase to explain why it's impossible for the user to delete those system-level catalog records (because their UUID's are generated when we first created the system, and those UUIDs are typically not seen by the users).